streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -606,6 +606,25 @@ public void onAssignment(Assignment assignment) {
             }
         }
         metadataWithInternalTopics = Cluster.empty().withPartitions(topicToPartitionInfo);
+        updateSubscribedTopics(assignment);
+    }
+
+
+    private void updateSubscribedTopics(Assignment assignment) {
+        if (streamThread != null && streamThread.builder.sourceTopicPattern() != null) {
```

Comment:
How could `streamThread` be `null` -- `null` would indicate a bug and we might want to fail hard on that instead of masking it with this check?